pypeit/utils.py

```diff
@@ -682,50 +713,58 @@ def bspline_profile(xdata, ydata, invvar, profile_basis, inmask = None, upper=5,
     yfit = np.zeros(ydata.shape)
     reduced_chi = 0.
 
-    if invvar.size == 1:
-        outmask = True
-    else:
-        outmask = np.ones(invvar.shape, dtype='bool')
-
-    if inmask is None:
-        inmask = (invvar > 0)
+    # TODO: Instanting these place-holder arrays can be expensive.  Can we avoid doing this?
+    outmask = True if invvar.size == 1 else np.ones(invvar.shape, dtype=bool)
 
-    nin = np.sum(inmask)
-    msgs.info("Fitting npoly =" + "{:3d}".format(npoly) + " profile basis functions, nin=" + "{:3d}".format(nin) + " good pixels")
-    msgs.info("******************************  Iter  Chi^2  # rejected  Rel. fact   ******************************")
-    msgs.info("                              ----  -----  ----------  --------- ")
+    if ingpm is None:
+        ingpm = invvar > 0
 
+    if not quiet:
+        termwidth = 80-13
+        msgs.info('B-spline fit:')
+        msgs.info('    npoly = {0} profile basis functions'.format(npoly))
+        msgs.info('    ngood = {0}/{1} measurements'.format(np.sum(ingpm), ingpm.size))
+        msgs.info(' {0:>4}  {1:>8}  {2:>7}  {3:>6} '.format(
+                    'Iter', 'Chi^2', 'N Rej', 'R. Fac').center(termwidth, '*'))
+        hlinestr = ' {0}  {1}  {2}  {3} '.format('-'*4, '-'*8, '-'*7, '-'*6)
+        nullval = '  {0:>8}  {1:>7}  {2:>6} '.format('-'*2, '-'*2, '-'*2)
+        msgs.info(hlinestr.center(termwidth))
 
-    maskwork = outmask & inmask & (invvar > 0)
+    maskwork = outmask & ingpm & (invvar > 0)
     if not maskwork.any():
         msgs.error('No valid data points in bspline_profile!.')
-    else:
-        # Init bspline class
-        sset = pydl.bspline(xdata[maskwork], nord=nord, npoly=npoly, bkpt=bkpt, fullbkpt=fullbkpt,
-                                funcname='Bspline longslit special', **kwargs_bspline)
-        if maskwork.sum() < sset.nord:
+
+    # Init bspline class
+#    sset = pydl.bspline(xdata[maskwork], nord=nord, npoly=npoly, bkpt=bkpt, fullbkpt=fullbkpt,
+#                            funcname='Bspline longslit special', **kwargs_bspline)
+    sset = bspline.bspline(xdata[maskwork], nord=nord, npoly=npoly, bkpt=bkpt, fullbkpt=fullbkpt,
+                           funcname='Bspline longslit special', **kwargs_bspline)
+    if maskwork.sum() < sset.nord:
+        if not quiet:
             msgs.warn('Number of good data points fewer than nord.')
-            exit_status = 4
-            return sset, outmask, yfit, reduced_chi, exit_status
+        # TODO: Why isn't maskwork returned?
```

Comment:
Because no rejection was performed, see below. Rejection is done with djs_reject.

Reply:
Okay, thanks.